[10.x] Solved the problem of incorrect parameter definition.

adideas opened the pull request:
My edits will help Laravel auto-detect the parameter correctly

```php
// api.php
Route::apiResource('accounts', 'Account\TestAccountController');
```
```php
class TestAccountController extends Controller {
   public function __construct()
   {
      $this->authorizeResource(TestAccount::class);
   }
      
   public function show(TestAccount $account)
   {
      return $account;
      // "message": "This action is unauthorized.",
      // "exception": "Symfony\\Component\\HttpKernel\\Exception\\AccessDeniedHttpException",
   }
}
```
```php
class AuthServiceProvider extends ServiceProvider {
   public function boot()
   {
      Gate::before(
         function (User $user, string $action, array $options) {
            // FOR TEST

            dd($action, $options);
            // return
            // "view"
            // [ ]       <- this problem (POLICY NOT RUN)
         }
      );
   }
}
```

If you do that, then everything works as it should. But the problem is auto-detect!

```php
class TestAccountController extends Controller {
   public function __construct()
   {
      $this->authorizeResource(TestAccount::class, 'account');
   }
```
My edits will help Laravel auto-detect the parameter correctly -> Laravel listens to the parameter specified in Route

```php
// before (It doesn't work for this example)
Str::snake(class_basename($model))
// after (Laravel listens to the parameter specified in Route)
Route::getCurrentRoute()->parameterNames
```
Please see what I have done in the file!

adideas committed:
Solved the problem of incorrect parameter definition. (AuthorizesRequests)

By default, the parameter should listen to the route parameter, not the model name.  Documentation [Authorizing Resource Controllers](https://laravel.com/docs/9.x/authorization#authorizing-resource-controllers) recommends specifying the parameter initially. The second parameter is optional. Ларавел will figure it out for you. In 99% of cases it works. I was lucky to be in the 1%. Below is a complete description of the problem.

StyleShit commented on src/Illuminate/Foundation/Auth/Access/AuthorizesRequests.php:
```diff
@@ -84,9 +84,9 @@ public function authorizeResource($model, $parameter = null, array $options = []
+        $parameter = $parameter ?: (Route::getCurrentRoute()->parameterNames ?: Str::snake(class_basename($model)));
 
-        $parameter = $parameter ?: Str::snake(class_basename($model));
+        $parameter = $parameter = is_array($parameter) ? implode(',', $parameter) : $parameter;
```
```suggestion
        $parameter = is_array($parameter) ? implode(',', $parameter) : $parameter;
```

adideas replied:
I did as you said. Please check.

StyleShit commented on src/Illuminate/Foundation/Auth/Access/AuthorizesRequests.php:
```diff
@@ -84,9 +84,9 @@ public function authorizeResource($model, $parameter = null, array $options = []
         $model = is_array($model) ? implode(',', $model) : $model;
 
-        $parameter = is_array($parameter) ? implode(',', $parameter) : $parameter;
+        $parameter = $parameter ?: (Route::getCurrentRoute()->parameterNames ?: Str::snake(class_basename($model)));
```
Maybe extract the nested conditional to another variable to make it more readable?

adideas replied:
I did as you said. Please check.

adideas committed:
Authorize a resource adding a variable to the function body for better code readability

Authorize a resource adding a variable to the function body for better code readability

adideas committed:
Removed the typo. $parameter = $parameter =

adideas committed:
Typo add use

adideas committed:
RuntimeException: A facade root has not been set.

driesvints commented:
There's no need to send this in twice as 9.x will be merged into master.
